bootstrap/src/main/java/jenkins/bootstrap/Bootstrap.java

```diff
@@ -0,0 +1,313 @@
+    private ClassLoader coreClassLoader;
+    /**
+     * JENKINS_HOME
+     */
+    private File home;
+
+    /**
+     * Creates the sole instance of {@link jenkins.model.Jenkins} and register it to the {@link ServletContext}.
+     */
+    public void contextInitialized(ServletContextEvent event) {
+        context = event.getServletContext();
+        context.setAttribute(Bootstrap.class.getName(),this);   // allow discovery of this
+        try {
+
+            final FileAndDescription describedHomeDir = getHomeDir(event);
+            home = describedHomeDir.file.getAbsoluteFile();
+            home.mkdirs();
+            System.out.println("Jenkins home directory: "+ home +" found at: "+describedHomeDir.description);
+
+            // check that home exists (as mkdirs could have failed silently), otherwise throw a meaningful error
+            if (!home.exists())
+                throw new Error("Invalid JENKINS_HOME: "+ home);
+
+            recordBootAttempt(home);
+
+            coreClassLoader = buildCoreClassLoader();
+            Thread.currentThread().setContextClassLoader(coreClassLoader);
+
+            bootLogics = loadBootLogics(coreClassLoader);
+
+            for (BootLogic b : bootLogics) {
+                b.contextInitialized(event);
+            }
+        } catch (Error | RuntimeException e) {
+            LOGGER.log(SEVERE, "Failed to initialize Jenkins",e);
+            throw e;
+        } catch (Exception e) {
+            LOGGER.log(SEVERE, "Failed to initialize Jenkins",e);
+            throw new Error(e);
+        }
+    }
+
+    public void contextDestroyed(ServletContextEvent event) {
+        for (BootLogic b : bootLogics) {
+            b.contextDestroyed(event);
+        }
+    }
+
+    /**
+     * {@link BootLogic}s that were found, in the order of preference.
+     */
+    public List<BootLogic> getBootLogics() {
+        return Collections.unmodifiableList(bootLogics);
+    }
+
+    /**
+     * ClassLoader that loads Jenkins core.
+     */
+    public ClassLoader getCoreClassLoader() {
+        return coreClassLoader;
+    }
+
+    /**
+     * Location of JENKINS_HOME.
+     */
+    public File getHome() {
+        return home;
+    }
+
+    private ClassLoader buildCoreClassLoader() throws IOException {
+        List<URL> urls = new ArrayList<>();
+        List<DependenciesTxt> overrides = new ArrayList<>();
+
+        DependenciesTxt core = new DependenciesTxt(getClass().getClassLoader().getResourceAsStream("dependencies.txt"));
+
+        File[] plugins = new File(home, "plugins").listFiles();
+        if (plugins!=null) {
+            for (File p : plugins) {
+                String n = p.getName();
+                if (p.isFile() && (n.endsWith(".jpi") || n.endsWith(".hpi"))) {
+                    try {
+                        new Plugin(p).process(core, urls, overrides);
+                    } catch (IOException e) {
+                        LOGGER.log(WARNING, "Skipping "+p, e);
```

Comment:
I think you misread the code here. The log message in question is issued when it's a plugin but the processing failed for unexpected reason (for example a corrupt jar file)